### PR TITLE
style(Container|Divider|Header|Icon): propTypes cleanups and typings update

### PR DESCRIPTION
**Pull request by layershifter**
This PR is part of work for removing propTypes in production bundle (#524, #731).
Also, cleanups and updates typings for #1072.

Affects following components:

- `Container`
- `Divider`
- `Header`
- `Icon`

### CODE REVIEW

File: src/elements/Header/Header.js [hunk 1]
```diff
@@ -80,29 +95,43 @@ function Header(props) {
...
 Header.propTypes = {
   /** An element type to render as (string or function). */
   as: customPropTypes.as,
 
-  /** Additional classes. */
-  className: PropTypes.string,
+  /** Attach header  to other content, like a segment. */
+  attached: PropTypes.oneOfType([
+    PropTypes.bool,
+    PropTypes.oneOf('top', 'bottom'),
```
Comment (levithomason) [inline src/elements/Header/Header.js]:
```diff
-PropTypes.oneOf('top', 'bottom'),
+PropTypes.oneOf(['top', 'bottom']),
```

File: src/elements/Header/index.d.ts [hunk 2]
```diff
@@ -1,17 +1,22 @@
-import { SemanticCOLORS, SemanticFLOATS, SemanticSIZES, SemanticTEXTALIGNMENTS } from '../..';
 import * as React from 'react';
+import {
+  SemanticCOLORS,
+  SemanticFLOATS,
+  SemanticTEXTALIGNMENTS
+} from '../..';
+
+type HEADER_SIZES = 'mini' | 'tiny' | 'small' | 'medium' | 'large' | 'huge'
```
Comment (levithomason) [inline src/elements/Header/index.d.ts]: Looks like this is only used in one place.  Suggest inlining it.

File: src/elements/Icon/index.d.ts [hunk 3]
```diff
@@ -71,4 +78,4 @@ interface IconGroupProps {
   size?: SemanticSIZES;
```
Comment (levithomason) [inline src/elements/Icon/index.d.ts]: I believe this should be `ICON_SIZES`, correct?

**Review by levithomason — CHANGES_REQUESTED** [review]

**Comment by layershifter** [comment]
TODO: update `content?: any;` to `React.Node` (`customPropTypes.contentShorthand`).

File: src/elements/Icon/index.d.ts (hunk 3 above)
Comment (layershifter) [reply src/elements/Icon/index.d.ts]: Yep, I also update Group's `propTypes`.

**Comment by layershifter** [comment]
We need merge #1165 before this, #1165 contains `implementsTextAlignProp` test update.

**Review by levithomason — APPROVED** [review]

**Comment by levithomason** [comment]
Merged #1165, fixed conflicts and unblocked.  Will merge on pass.

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.64.1`.

**Pull request merged**